### PR TITLE
Block API endpoints access to users when their password has been invalidated by a superuser

### LINKED ISSUE #4473: Add validated_password flag to accounts
- All accounts start with `validated_password: true` - superuser will manually switch them to `false` (for example based on account not changing password for x days after the new password constraints come into effect - using “last date of password change” from the point mentioned above)
- Account with `validated_password: false` will have restricted access to some APIs - handled by kobocat
- Account with `validated_password: false` will see a screen instead of KPI interface (handled by other issue) 
- Updating password should automatically switch the flag to `true`
- We need to build something into kpi so that whenever the superuser sets `validated_password: false` for an account, kpi will somehow flag that account in kobocat so that it cannot do anything except collect data (and reverse `validated_password: true` should work too, and unlock account)
- This flag should be exposted to frontend (in `/me` endpoint?)

### PR DESCRIPTION
**Pull request by noliveleger**
## Related issues

Closes #4473

### CODE REVIEW

**Review by JacquelineMorrissette — CHANGES_REQUESTED** [review]

File: kpi/serializers/current_user.py [hunk 1]
```diff
@@ -91,7 +91,7 @@ def get_validated_password(self, obj):
             extra_details = obj.extra_details
         except obj.extra_details.RelatedObjectDoesNotExist:
             # if user has not extra details, admin has not been able to set
-            # `validated_password`. Let's consider it True.
+            # `validated_password`. Let's consider it as True.
```
Comment (jnm) [inline kpi/serializers/current_user.py]: I'll revise this comment

**Review by jnm — APPROVED** [review]

File: kobo/apps/accounts/tests/test_social_account.py [hunk 2]
```diff
@@ -13,7 +13,7 @@ def test_list(self):
         account1 = baker.make('socialaccount.SocialAccount', user=self.user)
         account2 = baker.make('socialaccount.SocialAccount')
         # Auth, Count, Queryset
-        with self.assertNumQueries(3):
+        with self.assertNumQueries(4):
```
[the remark this answers is not in the document]
Comment (jnm) [reply kobo/apps/accounts/tests/test_social_account.py:16]: I'd like to see this too, but `FuzzyInt` isn't a built-in thing and we never introduced any implementation of it into the code base. We could do it in a separate PR

[the remark this answers is not in the document]
Comment (jnm) [reply kobo/apps/accounts/tests/test_social_account.py:16]: Oh, whoops. I didn't realize you'd already added it :innocent:

**Review by jnm — CHANGES_REQUESTED** [review]
When I try to unset `validated_password ` in the admin interface, it blocks me saying that fields are required—but they shouldn't be:
image

**Comment by jnm** [comment]
Discovered by accident: if you become logged out somehow and then attempt to change your password, you get a 500 error: `NotImplementedError: Django doesn't provide a DB representation for AnonymousUser.` It comes from https://github.com/kobotoolbox/kpi/blob/db36eb086fb5fc4af856f99ed24bb7e4e4079b0a/kpi/serializers/current_user.py#L148

File: kobo/apps/accounts/tests/test_social_account.py (hunk 2 above)
[the remark this answers is not in the document]
Comment (noliveleger) [reply kobo/apps/accounts/tests/test_social_account.py:16]: Yep, handled as a separate PR: #4606

File: kobo/apps/organizations/tests/test_organizations_api.py [hunk 3]
```diff
@@ -42,7 +42,7 @@ def test_list(self):
         self._insert_data()
         organization2 = baker.make(Organization, id='org_abcd123')
         organization2.add_user(user=self.user, is_admin=True)
-        with self.assertNumQueries(2):
+        with self.assertNumQueries(3):
```
[the remark this answers is not in the document]
Comment (noliveleger) [reply kobo/apps/organizations/tests/test_organizations_api.py:45]: Yep, handled as a separate PR: #4606

File: kobo/apps/organizations/tests/test_organizations_api.py [hunk 4]
```diff
@@ -61,7 +61,7 @@ def test_api_returns_org_data(self):
     def test_update(self):
         self._insert_data()
         data = {'name': 'edit'}
-        with self.assertNumQueries(4):
+        with self.assertNumQueries(5):
```
[the remark this answers is not in the document]
Comment (noliveleger) [reply kobo/apps/organizations/tests/test_organizations_api.py:64]: Yep, handled as a separate PR: #4606

**Review by jnm — APPROVED** [review]

**Pull request merged**